### ISSUE TITLE
about the training time?

### ISSUE THREAD

**Issue by xibaicai** [gligen/GLIGEN#20]
Hi, thanks your good work.
A few small questions about the implementation:
1. How long did your grounded with text experiment take to converge after training?How do I convert the 100K iterations you mentioned into epoch numbers? I don't know how to evaluate how long it takes to converge on my custom dataset

2. why did you set max_boxes_per_data=30? Does it not work well when it is bigger? Any experience here?

3. Why do you have to go through the "scale to 0-1" operation first when processing the box coordinates? What is the reason for this step?

4. When getting the Box Grounding Tokens, you mentioned in your paper that 
`"we choose "EOS" token feature at this layer as our grounded text embedding ." `
Is the approach here to take the last one of the 77 tokens directly? For example the **feature.shape=(77, 768)** extracted by CLIPEm bedder, then take the slice **feature[-1, :]**, am I doing it right?

I would like to get your answer!

**Comment by Yuheng-Li** [comment]
1) 100k iterations generally takes 1-2 days. For COCO we train our model using 100k (which is around 80 epoches), but we train 500k for GoldG+SBU+CC3M+O365 datasets. We actually did not notice that our model is fully converged, we recommend to try at least 500k if your custom dataset is large.    

2) we empirically found that most images have bounding boxes less than 30 in training data; and in practice, users usually do not draw such many boxes. Our model should still work if you set it with larger number, say 100, but this will take more memory during training.

3) we normalize to 0-1 simply considering the sin and cos operation. (it should be fine if you use raw coordinates; some nerf paper doing this). In our dataset, we will crop the image into 512*512 and recalculate boxes coordinates  if the training image is not square

**Comment by xibaicai** [comment]
Thank you for your sincere reply. 
Reply 1: The same phenomenon is that the loss keeps decreasing with no trend of convergence while I am training.

There is another evaluation question that has been bothering me for a long time. **During the training process, how can I see in real time whether the model has adapted to the grounding condition we introduced?** I am currently just looking at a few images of the intermediate sampling results saved during the training process, and this method of looking at a few images with the naked eye is very unintuitive.

Any good suggestions, thanks!

**Comment by Bailey-24** [comment]
why gligen is always output 512*512, how to output 640*480
